packages/models/src/huggingface/text-generation.ts

```diff
@@ -0,0 +1,210 @@
+import { HttpError, POST } from '@axflow/models/shared';
+
+// HuggingFace has the concept of a task. This code supports the "textGeneration" task.
+// https://huggingface.co/models?pipeline_tag=text-generation
+
+// https://huggingface.co/docs/api-inference/quicktour#running-inference-with-api-requests
+const HF_MODEL_API_URL = 'https://api-inference.huggingface.co/models/';
+
+function headers(accessToken?: string) {
+  const headers: Record<string, string> = {
+    accept: 'application/json',
+    'content-type': 'application/json',
+  };
+  if (typeof accessToken === 'string') {
+    headers.authorization = `Bearer ${accessToken}`;
+  }
+  return headers;
+}
+
+export namespace HfChatTypes {
+  // https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task
+  export type Request = {
+    model: string;
+    stream?: boolean;
+    inputs: string;
+    parameters?: {
+      top_k?: number;
+      top_p?: number;
+      temperature?: number;
+      repetition_penalty?: number;
+      max_new_tokens?: number;
+      // In seconds
+      max_time?: number;
+      return_full_text?: boolean;
+      num_return_sequences?: number;
+      do_sample?: boolean;
+    };
+    options?: {
+      use_cache?: boolean;
+      wait_for_model?: boolean;
+    };
+  };
+
+  export type RequestOptions = {
+    accessToken?: string;
+    apiUrl?: string;
+    fetch?: typeof fetch;
+  };
+
+  export type GeneratedText = {
+    generated_text: string;
+  };
+
+  // https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task
+  export type Response = GeneratedText | GeneratedText[];
+
+  // Best documentation I could find: https://huggingface.co/docs/huggingface_hub/main/en/package_reference/inference_client#huggingface_hub.inference._text_generation.TextGenerationStreamResponse
+  // I would like to find more formal documentation of their streaming API if we can.
+  export type Chunk = {
+    token: {
+      id: number;
+      text: string;
+      logprob: number;
+      special: boolean;
+    };
+    generated_text: string;
+    // Observed this but cannot find documentation
```

Comment:
They seem to mention it in the streaming guide: https://huggingface.co/docs/text-generation-inference/conceptual/streaming#streaming-with-python

That said, the docs are poor. In either case, it seems it can be non-null in certain cases which means the typing here is wrong. I'd either spend more time to understand what the possible values are and type them appropriately, or possibly omit it for now if it's not a commonly used parameter.